inc/saitypes.h

```diff
@@ -802,6 +802,19 @@ typedef sai_status_t (*sai_bulk_object_remove_fn)(
         _In_ sai_bulk_op_error_mode_t mode,
         _Out_ sai_status_t *object_statuses);
 
+typedef enum _sai_stats_mode_t
+{
+    /**
+     * @brief Read statistics
+     */
+    SAI_STATS_MODE_READ,
+
+    /**
+     * @brief Read and clear after reading
+     */
+    SAI_STATS_MODE_READ_AND_CLEAR,
```

Comment:
for this to be atomic operation, actual asic needs to support such capability and maybe not all asic can support that

for example if asic dont support that, there is no way to guarantee atomic on that, in port stat when port is running like 100gbps, then after just reading stats all the byte counters will gain a lot until clear will be called